### PR TITLE
feat(csi-node): allow listening on IPv6 Pod IPs

### PR DESCRIPTION
**Pull request by michaelbeaumont**
See https://github.com/openebs/mayastor/issues/1731

### CODE REVIEW

File: control-plane/csi-driver/src/bin/node/main_.rs [hunk 1]
```diff
@@ -427,21 +442,35 @@ async fn check_ana_and_label_node(
...
     registration_enabled: bool,
 ) -> anyhow::Result<SocketAddr> {
-    let grpc_endpoint = if grpc_endpoint.contains(':') {
-        grpc_endpoint.to_string()
+    let grpc_endpoint_src = matches.value_source("grpc-endpoint");
+    let grpc_ip = matches.get_one::<IpAddr>("grpc-ip");
+    let grpc_port = matches.get_one::<u16>("grpc-port");
+    let grpc_endpoint_socket_addr = if grpc_endpoint_src
+        .unwrap_or(clap::parser::ValueSource::DefaultValue)
+        == clap::parser::ValueSource::DefaultValue
```
Comment (tiagolobocastro) [inline control-plane/csi-driver/src/bin/node/main_.rs]: What about making `grpc-endpoint` optional, and adding a clap conflicts with `grpc-ip` and `grpc-port` ? 
Would it be more intuitive?

Comment (michaelbeaumont) [reply control-plane/csi-driver/src/bin/node/main_.rs]: I added a `conflicts_with_all` for both the args. Does that help with using `matches` here somehow? I'm no expert when it comes to clap's builder, feel free to make a suggestion :)

Comment (tiagolobocastro) [reply control-plane/csi-driver/src/bin/node/main_.rs]: awesome, that's it, now it complains straight away:

```
❯ csi-node --node-name a --grpc-endpoint 127.0.0.1 --grpc-ip 127.0.0.1 --grpc-port 1080                                                                                                                                                  ─╯
error: the argument '--grpc-endpoint <ENDPOINT>' cannot be used with:
  --grpc-ip <GRPC_IP>
  --grpc-port <GRPC_PORT>
```

Also in addition, you can remove the default value from `grpc-endpoint`.

**Review by tiagolobocastro — APPROVED** [review]
Looks like DCO is failing, please amend commit with signature.
Also I suggest not to add the github smart link on the PR description. We use bors with merge, including PR description so everything a branch is pushed there's a github event on the mentioned ticket. In case someone is push a lot of branches for some reason it floods the github ticket.

**Comment by tiagolobocastro** [comment]
bors d+

**Comment by michaelbeaumont** [comment]
@tiagolobocastro so only put the link in the commit message?

**Comment by tiagolobocastro** [comment]
> @tiagolobocastro so only put the link in the commit message?

I would suggest leaving it as a comment instead, though now that I think about it, not sure github would then auto close issue in this case.
So feel free to do whatever makes more sense to you.

**Comment by michaelbeaumont** [comment]
@tiagolobocastro sorry wasn't clear, I think I can put it in the _merged_ commit message without flooding the issue. Hmm actually I guess bors controls that. So there's just never a link back to an issue anywhere?

File: control-plane/csi-driver/src/bin/node/main_.rs (hunk 1 above)
Comment (michaelbeaumont) [reply control-plane/csi-driver/src/bin/node/main_.rs]: done! simplified the validation code

**Review by tiagolobocastro — APPROVED** [review]

**Comment by michaelbeaumont** [comment]
bors try

**Comment by tiagolobocastro** [comment]
sigh:
```
[2024-10-09T19:16:56.607Z] OpenTelemetry trace error occurred. Exporter otlp encountered the following error(s): the grpc server returns error (The service is currently unavailable): , detailed error message: error trying to connect: tcp connect error: Connection refused (os error 111)
```
I'm tempted to disable jaeger tracing from the tests...

bors try

**Comment by michaelbeaumont** [comment]
bors merge

**Pull request merged**